types/inferschematype.d.ts

```diff
@@ -75,7 +75,7 @@ type IsPathRequired<P, TypeKey extends TypeKeyBaseType> =
       ? P extends { default: undefined }
         ? false
         : true
-    : P extends PathWithTypePropertyBaseType<TypeKey>
+    : P extends (Record<TypeKey, NumberSchemaDefinition | StringSchemaDefinition | BooleanSchemaDefinition | DateSchemaDefinition>)
```

Comment:
I assumed not all of typeKey's types should be allowed to be remove the optional flag on that path.

So I set it to these 4. Because I think they're the only ones that should be able to provide defaults.

Reply:
So if you don't want the types to be limited, I'll revert the whole commit.